### ISSUE TITLE
Pip install dev dependencies fails on python 2.7

### ISSUE THREAD

**Issue by aop** [mehcode/python-xmlsec#24]
When running `pip install -e ".[test]"`

```
$ pip install -e ".[test]"
Obtaining file:///home/myuser/python-xmlsec
  Running setup.py  egg_info for package from file:///home/myuser/python-xmlsec

  Installing extra requirements: 'test'
Requirement already satisfied (use --upgrade to upgrade): lxml>=3.0 in ./env/lib/python2.7/site-packages (from xmlsec==0.3.1)
Requirement already satisfied (use --upgrade to upgrade): pytest in ./env/lib/python2.7/site-packages (from xmlsec==0.3.1)
Installing collected packages: xmlsec
  Running setup.py develop for xmlsec

    skipping 'src/xmlsec/constants.c' Cython extension (up-to-date)
    building 'xmlsec.constants' extension
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DXMLSEC_NO_CRYPTO_DYNAMIC_LOADING=1 -DXMLSEC_OPENSSL_098=1 -DXMLSEC_NO_GOST=1 -D__XMLSEC_FUNCTION__=__FUNCTION__ -DXMLSEC_CRYPTO_OPENSSL=1 -DXMLSEC_NO_XKMS=1 -DXMLSEC_NO_SIZE_T= -DXMLSEC_CRYPTO=openssl -Isrc -I/home/myuser/python-xmlsec/env/local/lib/python2.7/site-packages/lxml/includes -I/home/myuser/python-xmlsec/env/local/lib/python2.7/site-packages/lxml -I/usr/include/xmlsec1 -I/usr/include/libxml2 -I/usr/include/python2.7 -c src/xmlsec/constants.c -o build/temp.linux-x86_64-2.7/src/xmlsec/constants.o -DXMLSEC_CRYPTO_OPENSSL=1
    src/xmlsec/constants.c: In function ‘initconstants’:
    src/xmlsec/constants.c:2468:61: error: ‘xmlSecTransformEcdsaSha1Id’ undeclared (first use in this function)
       __pyx_t_2 = ((PyObject *)__pyx_f_6xmlsec_9constants__mkti(xmlSecTransformEcdsaSha1Id)); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 105, __pyx_L1_error)
                                                                 ^
    src/xmlsec/constants.c:2468:61: note: each undeclared identifier is reported only once for each function it appears in
    src/xmlsec/constants.c:2480:61: error: ‘xmlSecTransformEcdsaSha224Id’ undeclared (first use in this function)
       __pyx_t_2 = ((PyObject *)__pyx_f_6xmlsec_9constants__mkti(xmlSecTransformEcdsaSha224Id)); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 106, __pyx_L1_error)
                                                                 ^
    src/xmlsec/constants.c:2492:61: error: ‘xmlSecTransformEcdsaSha256Id’ undeclared (first use in this function)
       __pyx_t_2 = ((PyObject *)__pyx_f_6xmlsec_9constants__mkti(xmlSecTransformEcdsaSha256Id)); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 107, __pyx_L1_error)
                                                                 ^
    src/xmlsec/constants.c:2504:61: error: ‘xmlSecTransformEcdsaSha384Id’ undeclared (first use in this function)
       __pyx_t_2 = ((PyObject *)__pyx_f_6xmlsec_9constants__mkti(xmlSecTransformEcdsaSha384Id)); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 108, __pyx_L1_error)
                                                                 ^
    src/xmlsec/constants.c:2516:61: error: ‘xmlSecTransformEcdsaSha512Id’ undeclared (first use in this function)
       __pyx_t_2 = ((PyObject *)__pyx_f_6xmlsec_9constants__mkti(xmlSecTransformEcdsaSha512Id)); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 109, __pyx_L1_error)
                                                                 ^
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    Complete output from command /home/myuser/python-xmlsec/env/bin/python -c "import setuptools, tokenize; __file__='/home/myuser/python-xmlsec/setup.py'; exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" develop --no-deps:
    running develop

running egg_info

writing dependency_links to xmlsec.egg-info/dependency_links.txt

writing requirements to xmlsec.egg-info/requires.txt

writing xmlsec.egg-info/PKG-INFO

writing top-level names to xmlsec.egg-info/top_level.txt

warning: manifest_maker: standard file '-c' not found



reading manifest file 'xmlsec.egg-info/SOURCES.txt'

reading manifest template 'MANIFEST.in'

writing manifest file 'xmlsec.egg-info/SOURCES.txt'

running build_ext

skipping 'src/xmlsec/constants.c' Cython extension (up-to-date)

building 'xmlsec.constants' extension

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DXMLSEC_NO_CRYPTO_DYNAMIC_LOADING=1 -DXMLSEC_OPENSSL_098=1 -DXMLSEC_NO_GOST=1 -D__XMLSEC_FUNCTION__=__FUNCTION__ -DXMLSEC_CRYPTO_OPENSSL=1 -DXMLSEC_NO_XKMS=1 -DXMLSEC_NO_SIZE_T= -DXMLSEC_CRYPTO=openssl -Isrc -I/home/myuser/python-xmlsec/env/local/lib/python2.7/site-packages/lxml/includes -I/home/myuser/python-xmlsec/env/local/lib/python2.7/site-packages/lxml -I/usr/include/xmlsec1 -I/usr/include/libxml2 -I/usr/include/python2.7 -c src/xmlsec/constants.c -o build/temp.linux-x86_64-2.7/src/xmlsec/constants.o -DXMLSEC_CRYPTO_OPENSSL=1

src/xmlsec/constants.c: In function ‘initconstants’:

src/xmlsec/constants.c:2468:61: error: ‘xmlSecTransformEcdsaSha1Id’ undeclared (first use in this function)

   __pyx_t_2 = ((PyObject *)__pyx_f_6xmlsec_9constants__mkti(xmlSecTransformEcdsaSha1Id)); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 105, __pyx_L1_error)

                                                             ^

src/xmlsec/constants.c:2468:61: note: each undeclared identifier is reported only once for each function it appears in

src/xmlsec/constants.c:2480:61: error: ‘xmlSecTransformEcdsaSha224Id’ undeclared (first use in this function)

   __pyx_t_2 = ((PyObject *)__pyx_f_6xmlsec_9constants__mkti(xmlSecTransformEcdsaSha224Id)); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 106, __pyx_L1_error)

                                                             ^

src/xmlsec/constants.c:2492:61: error: ‘xmlSecTransformEcdsaSha256Id’ undeclared (first use in this function)

   __pyx_t_2 = ((PyObject *)__pyx_f_6xmlsec_9constants__mkti(xmlSecTransformEcdsaSha256Id)); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 107, __pyx_L1_error)

                                                             ^

src/xmlsec/constants.c:2504:61: error: ‘xmlSecTransformEcdsaSha384Id’ undeclared (first use in this function)

   __pyx_t_2 = ((PyObject *)__pyx_f_6xmlsec_9constants__mkti(xmlSecTransformEcdsaSha384Id)); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 108, __pyx_L1_error)

                                                             ^

src/xmlsec/constants.c:2516:61: error: ‘xmlSecTransformEcdsaSha512Id’ undeclared (first use in this function)

   __pyx_t_2 = ((PyObject *)__pyx_f_6xmlsec_9constants__mkti(xmlSecTransformEcdsaSha512Id)); if (unlikely(!__pyx_t_2)) __PYX_ERR(0, 109, __pyx_L1_error)

                                                             ^

error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

----------------------------------------
Cleaning up...
Command /home/myuser/python-xmlsec/env/bin/python -c "import setuptools, tokenize; __file__='/home/myuser/python-xmlsec/setup.py'; exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" develop --no-deps failed with error code 1 in /home/myuser/python-xmlsec
Traceback (most recent call last):
  File "/home/myuser/python-xmlsec/env/bin/pip", line 11, in <module>
    sys.exit(main())
  File "/home/myuser/python-xmlsec/env/local/lib/python2.7/site-packages/pip/__init__.py", line 185, in main
    return command.main(cmd_args)
  File "/home/myuser/python-xmlsec/env/local/lib/python2.7/site-packages/pip/basecommand.py", line 161, in main
    text = '\n'.join(complete_log)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 40: ordinal not in range(128)
```

I'm running Linux Mint 17.2 64-bit, compiling with gcc version 4.8.4 (Ubuntu 4.8.4-2ubuntu1~14.04.1)

**Comment by bgaifullin** [comment]
Have you installed all requirements. I mean libxml2-dev libxmlsec1-dev. 
Seems like the main reason that compilation has been failed.

**Comment by bgaifullin** [comment]
@aop Could you test your case with this PR #37.

**Comment by bgaifullin** [comment]
The new version 1.0.1 was released. currently this issue is obsolete, since 1.0.1 is written on pure C.
@aop could you check with latest version of python-xmlsec and if it also does not work for you fire new issue. 
This I will close as obsolete.
